### PR TITLE
Use DataCite to mint DOIs

### PR DESCRIPTION
**Pull request by awead**
Updates the ezid-client gem to support the EZ API and adds the required credentials for authentication.

We can still continue to use the current EZID API by using the appropriate `EZID_HOST` and corresponding user, password, and handle. When the time comes to switch to using DataCite's EZ API, we specify the DateCite hostname, as well as corresponding user, password, and new handle.

Connected to #1237

### CODE REVIEW

File: app/services/doi_service.rb [hunk 1]
```diff
@@ -24,10 +24,10 @@ class DOIService
     'Other' => 'Other'
   }.freeze
 
-  def initialize(handle = ScholarSphere::Application.config.doi_handle,
-                 user = ScholarSphere::Application.config.doi_user,
-                 password = ScholarSphere::Application.config.doi_password)
-    @client = Ezid::Client.new(user: user, password: password)
+  # @note user, password, and host values for the Ezid::Client are obtained from environment variables
+  #   set in application.yml. The hostname can be either an EZID API service or the new DataCite EZ API.
+  def initialize(handle = ScholarSphere::Application.config.doi_handle)
```
Comment (carolyncole) [inline app/services/doi_service.rb:29]: @awead how does this transition to DataCite?  Does the EzidClient just work?

**Review by carolyncole — CHANGES_REQUESTED** [review]
@awead It would be nice to have a test that calls out to datacite to be certain that it will work when we need to transition.

File: app/services/doi_service.rb (hunk 1 above)
Comment (awead) [reply app/services/doi_service.rb:29]: The ezid-client gem supports both. Just changing the hostname is all that's needed.

**Comment by awead** [comment]
@cam156 should we have a test that's mocked via VCR/webmock?

**Comment by carolyncole** [comment]
@awead a vcr test seems like a good idea.  Did we need to do anything related to the fact that DataCite will not accept Book as a type?

**Comment by awead** [comment]
@cam156 RE: VCR, I agree it would be helpful, but I'm hesitant to do it now given the changes to their test instance. Also, the VCR test wouldn't catch these changes in the future. It would however, ensure that nothing was breaking on our side, so it's still worthwhile. I'd just rather wait until we know for sure what kind of request we'll need to send to them.

**Comment by awead** [comment]
@cam156 BTW, I have tested this branch locally using the current EZID host and credentials that are in production, and DOIs are getting minted. This same branch worked _last week_ before the apparent changes to DataCite's test instance. So we can be confident that this change does not break current production, but due to DataCite's changes, we'd likely have to make additional changes to mint DOI successfully.

That being said, this PR does provide "support" for both EZID and DataCite. It's just that the latter does not accept all our metadata in its present state.

**Review by carolyncole — APPROVED** [review]

**Pull request merged**